lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1233,6 +1281,32 @@ def configure_disks(self, vm_obj):
 
                 self.change_detected = True
 
+    def create_directory_for_datadisk(self, datacenter, datastore, vm_name, diskspec):
+        """
+        Helper function to create a directory for data disk
+        """
+        parent_dc = datacenter
+        datastore_name = datastore.name
+        # Try to create folder for virtual machine and disks inside datastore
+        path_on_ds = '[' + datastore_name + ']' + vm_name
+        try:
+            self.content.fileManager.MakeDirectory(name=path_on_ds,
```

Comment:
Are you sure this is OK ?
Have a look at: https://github.com/ansible/ansible/pull/33315

Reply:
You are right, as is it creates a file directly under the datastore which is similar to the issue you pointed out.  I don't think that just the datastore can be used as a valid path in the same manner though.  I believe the authors intent is to create unique vmdk names using the information from diskspec.  I did have to make this change to make it work though.

vmdk_file_name = path_on_ds + '/' + vm_name +  '/' + vm_name + '_' + disk_file_name + '.vmdk'